File: src/main/java/com/handwoong/everyonewaiter/exception/ErrorResponse.java
package com.handwoong.everyonewaiter.exception;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;
import org.springframework.http.ResponseEntity;

@Getter
@Builder
public class ErrorResponse {

    private final String error;

    private final String message;

    private final int status;

    private final String code;

    private final LocalDateTime timestamp = LocalDateTime.now();

    public static ResponseEntity<ErrorResponse> toResponseEntity(ErrorCode errorCode) {
        ErrorResponse errorResponse = ErrorResponse.builder()
                .error(errorCode.getStatus().name())
                .message(errorCode.getMessage())
                .status(errorCode.getStatus().value())
                .code(errorCode.name())
                .build();
        return ResponseEntity.status(errorCode.getStatus()).body(errorResponse);
    }
}

File: src/main/kotlin/com/handwoong/everyonewaiter/exception/ErrorResponse.kt
package com.handwoong.everyonewaiter.exception

import org.springframework.http.ResponseEntity
import java.time.LocalDateTime

data class ErrorResponse(
    val error: String,
    val message: String,
    val status: Int,
    val code: String,
    val timestamp: LocalDateTime,
) {

    companion object {
        fun of(errorCode: ErrorCode): ResponseEntity<ErrorResponse> {
            val errorResponse = ErrorResponse(
                error = errorCode.status.name,
                message = errorCode.message,
                status = errorCode.status.value(),
                code = errorCode.name,
                timestamp = LocalDateTime.now(),
            )
            return ResponseEntity.status(errorCode.status).body(errorResponse)
        }
    }

}
